Source/Core/VideoCommon/TextureCacheBase.cpp

```diff
@@ -897,6 +913,31 @@ TextureCache::TCacheEntryBase* TextureCache::AllocateTexture(const TCacheEntryCo
 	return g_texture_cache->CreateTexture(config);
 }
 
+TextureCache::TexCache::iterator TextureCache::RemoveTextureFromCache(TexCache::iterator t_iter)
+{
+	// Remove the cache entry for normal textures from textures_by_hash and textures_by_address
+	if (!t_iter->second->IsEfbCopy())
+	{
+		std::pair<TexCache::iterator, TexCache::iterator> iter_range = textures_by_hash.equal_range(t_iter->second->hash);
```

Comment:
A c++ map iterator remains valid for a very long time, usually as long as you don't delete the element. So you're able to persistently store the iterator of the hash_map within the TCacheEntryBase class. In this case, you don't need to search within this map at all.